Build failure with ffmpeg-4.x

### Steps to reproduce

Build with ffmpeg-4.x.

### Expected behaviour

Successful build.

### Actual behaviour

```
[443/1490] /usr/bin/x86_64-pc-linux-gnu-g++ -DCRL_FORCE_COMMON_QUEUE -DDESKTOP_APP_DISABLE_WAYLAND_INTEGRATION -DGI_INLINE -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_56 -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_56 -DKCOREADDONS_EXPORT="" -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_DEPRECATED_WARNINGS_SINCE=0x051500 -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_FROM_BYTEARRAY -DQT_NO_DEBUG -DQT_NO_KEYWORDS -DQT_OPENGLWIDGETS_LIB -DQT_OPENGL_LIB -DQT_SVG_LIB -DQT_WIDGETS_LIB -I/var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/SourceFiles -I/var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/lib_base -I/var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/lib_rpl -I/var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/lib_crl -I/var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/lib_ui -I/var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/lib_ui/emoji_suggestions -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/lib64/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtCore/6.7.3 -isystem /usr/include/qt6/QtCore/6.7.3/QtCore -isystem /usr/include/qt6/QtGui -isystem /usr/include/qt6/QtGui/6.7.3 -isystem /usr/include/qt6/QtGui/6.7.3/QtGui -isystem /usr/include/qt6/QtOpenGL -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtWidgets/6.7.3 -isystem /usr/include/qt6/QtWidgets/6.7.3/QtWidgets -isystem /usr/include/qt6/QtOpenGLWidgets -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtSvg -isystem /usr/include/qt6/QtDBus -isystem /var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/ThirdParty/expected/include -isystem /var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/ThirdParty/kcoreaddons/src/lib/io -isystem /var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/ThirdParty/kcoreaddons/src/lib/util -isystem /var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/cmake/external/kcoreaddons/headers/public -isystem /var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full_build/Telegram/lib_base/gen -isystem /var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full_build/gen -isystem /usr/include/cppgir -isystem /usr/include/cppgir/gi -isystem /usr/include/cppgir/override -isystem /usr/include/gio-unix-2.0 -isystem /usr/lib64/libffi/include -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full_build/Telegram/lib_ui/gen  -DNDEBUG -march=native -O2 -pipe -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs -fuse-ld=mold -std=gnu++20 -fPIC -fno-strict-aliasing -pipe -Wall -Wextra -Wno-unused-parameter -Wno-switch -Wno-maybe-uninitialized -Wno-missing-field-initializers -Wno-sign-compare -Wno-deprecated -pthread -MD -MT Telegram/CMakeFiles/lib_ffmpeg.dir/SourceFiles/ffmpeg/ffmpeg_utility.cpp.o -MF Telegram/CMakeFiles/lib_ffmpeg.dir/SourceFiles/ffmpeg/ffmpeg_utility.cpp.o.d -o Telegram/CMakeFiles/lib_ffmpeg.dir/SourceFiles/ffmpeg/ffmpeg_utility.cpp.o -c /var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/SourceFiles/ffmpeg/ffmpeg_utility.cpp
FAILED: Telegram/CMakeFiles/lib_ffmpeg.dir/SourceFiles/ffmpeg/ffmpeg_utility.cpp.o 
/usr/bin/x86_64-pc-linux-gnu-g++ -DCRL_FORCE_COMMON_QUEUE -DDESKTOP_APP_DISABLE_WAYLAND_INTEGRATION -DGI_INLINE -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_56 -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_56 -DKCOREADDONS_EXPORT="" -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_DEPRECATED_WARNINGS_SINCE=0x051500 -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_FROM_BYTEARRAY -DQT_NO_DEBUG -DQT_NO_KEYWORDS -DQT_OPENGLWIDGETS_LIB -DQT_OPENGL_LIB -DQT_SVG_LIB -DQT_WIDGETS_LIB -I/var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/SourceFiles -I/var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/lib_base -I/var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/lib_rpl -I/var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/lib_crl -I/var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/lib_ui -I/var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/lib_ui/emoji_suggestions -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/lib64/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtCore/6.7.3 -isystem /usr/include/qt6/QtCore/6.7.3/QtCore -isystem /usr/include/qt6/QtGui -isystem /usr/include/qt6/QtGui/6.7.3 -isystem /usr/include/qt6/QtGui/6.7.3/QtGui -isystem /usr/include/qt6/QtOpenGL -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtWidgets/6.7.3 -isystem /usr/include/qt6/QtWidgets/6.7.3/QtWidgets -isystem /usr/include/qt6/QtOpenGLWidgets -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtSvg -isystem /usr/include/qt6/QtDBus -isystem /var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/ThirdParty/expected/include -isystem /var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/ThirdParty/kcoreaddons/src/lib/io -isystem /var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/ThirdParty/kcoreaddons/src/lib/util -isystem /var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/cmake/external/kcoreaddons/headers/public -isystem /var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full_build/Telegram/lib_base/gen -isystem /var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full_build/gen -isystem /usr/include/cppgir -isystem /usr/include/cppgir/gi -isystem /usr/include/cppgir/override -isystem /usr/include/gio-unix-2.0 -isystem /usr/lib64/libffi/include -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full_build/Telegram/lib_ui/gen  -DNDEBUG -march=native -O2 -pipe -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs -fuse-ld=mold -std=gnu++20 -fPIC -fno-strict-aliasing -pipe -Wall -Wextra -Wno-unused-parameter -Wno-switch -Wno-maybe-uninitialized -Wno-missing-field-initializers -Wno-sign-compare -Wno-deprecated -pthread -MD -MT Telegram/CMakeFiles/lib_ffmpeg.dir/SourceFiles/ffmpeg/ffmpeg_utility.cpp.o -MF Telegram/CMakeFiles/lib_ffmpeg.dir/SourceFiles/ffmpeg/ffmpeg_utility.cpp.o.d -o Telegram/CMakeFiles/lib_ffmpeg.dir/SourceFiles/ffmpeg/ffmpeg_utility.cpp.o -c /var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/SourceFiles/ffmpeg/ffmpeg_utility.cpp
/var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/SourceFiles/ffmpeg/ffmpeg_utility.cpp: In function ‘FFmpeg::SwresamplePointer FFmpeg::MakeSwresamplePointer(uint64_t, AVSampleFormat, int, uint64_t, AVSampleFormat, int, SwresamplePointer*)’:
/var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/SourceFiles/ffmpeg/ffmpeg_utility.cpp:567:37: error: request for member ‘get’ in ‘existing’, which is of pointer type ‘FFmpeg::SwresamplePointer*’ {aka ‘std::unique_ptr<SwrContext, FFmpeg::SwresampleDeleter>*’} (maybe you meant to use ‘->’ ?)
  567 |                 existing ? existing.get() : nullptr,
      |                                     ^~~
/var/tmp/portage/net-im/telegram-desktop-5.7.2/work/tdesktop-5.7.2-full/Telegram/SourceFiles/ffmpeg/ffmpeg_utility.cpp:581:9: error: ‘error’ was not declared in this scope; did you mean ‘perror’?
  581 |         error = AvErrorWrap(swr_init(result));
      |         ^~~~~
      |         perror
```
See https://bugs.gentoo.org/942770 for detailed info.

### Operating system

Gentoo

### Version of Telegram Desktop

5.7.2

### Installation source

Other (unofficial) source

### Crash ID

_No response_

### Logs

```shell
https://bugs.gentoo.org/942770
```

--- a/Telegram/SourceFiles/ffmpeg/ffmpeg_utility.cpp
+++ b/Telegram/SourceFiles/ffmpeg/ffmpeg_utility.cpp
@@ -546,9 +546,10 @@ SwresamplePointer MakeSwresamplePointer(
 	}
 
 	// Initialize audio resampler
+	AvErrorWrap error;
 #if DA_FFMPEG_NEW_CHANNEL_LAYOUT
 	auto result = (SwrContext*)nullptr;
-	auto error = AvErrorWrap(swr_alloc_set_opts2(
+	error = AvErrorWrap(swr_alloc_set_opts2(
 		&result,
 		dstLayout,
 		dstFormat,
@@ -564,7 +565,7 @@ SwresamplePointer MakeSwresamplePointer(
 	}
 #else // DA_FFMPEG_NEW_CHANNEL_LAYOUT
 	auto result = swr_alloc_set_opts(
-		existing ? existing.get() : nullptr,
+		existing ? existing->get() : nullptr,
 		dstLayout,
 		dstFormat,
 		dstRate,

--- a/Telegram/SourceFiles/media/audio/media_audio_local_cache.cpp
+++ b/Telegram/SourceFiles/media/audio/media_audio_local_cache.cpp
@@ -44,7 +44,13 @@ constexpr auto kFrameSize = 4096;
 		return {};
 	}
 
+
+#if LIBAVFORMAT_VERSION_INT >= AV_VERSION_INT(58, 79, 100)
 	auto inCodec = (const AVCodec*)nullptr;
+#else
+	auto inCodec = (AVCodec*)nullptr;
+#endif
+
 	const auto streamId = av_find_best_stream(
 		input.get(),
 		AVMEDIA_TYPE_AUDIO,
@@ -152,10 +158,10 @@ constexpr auto kFrameSize = 4096;
 		inCodecContext->sample_rate,
 		&outCodecContext->ch_layout,
 #else // DA_FFMPEG_NEW_CHANNEL_LAYOUT
-		&inCodecContext->channel_layout,
+		inCodecContext->channel_layout,
 		inCodecContext->sample_fmt,
 		inCodecContext->sample_rate,
-		&outCodecContext->channel_layout,
+		outCodecContext->channel_layout,
 #endif // DA_FFMPEG_NEW_CHANNEL_LAYOUT
 		outCodecContext->sample_fmt,
 		outCodecContext->sample_rate);

--- a/Telegram/SourceFiles/ui/controls/round_video_recorder.cpp
+++ b/Telegram/SourceFiles/ui/controls/round_video_recorder.cpp
@@ -341,10 +341,10 @@ bool RoundVideoRecorder::Private::initAudio() {
 		&_swrContext);
 #else // DA_FFMPEG_NEW_CHANNEL_LAYOUT
 	_swrContext = MakeSwresamplePointer(
-		&_audioCodec->channel_layout,
+		_audioCodec->channel_layout,
 		AV_SAMPLE_FMT_S16,
 		_audioCodec->sample_rate,
-		&_audioCodec->channel_layout,
+		_audioCodec->channel_layout,
 		_audioCodec->sample_fmt,
 		_audioCodec->sample_rate,
 		&_swrContext);

